File: laboratories/ppo/exercises/lab10/java/lab10/Router.java
package lab10;

import java.util.HashMap;
import java.util.Map;

public class Router {
    private final Map<String, Route> routes = new HashMap<>();

    public Router(Route[] routing) {
        for (Route route : routing) {
            routes.put(route.getPath(), route);
        }
    }

    public Route match(String path) {
        return routes.get(path);
    }
}

File: laboratories/ppo/exercises/lab10/php/src/Router.php
<?php

namespace Ppo\Lab10;

class Router
{
    /** @var array<Route>  */
    protected array $routing;

    public function __construct(array $routing)
    {
        $this->routing = array_combine(
            array_map(fn(Route $route): string => $route->route, $routing),
            $routing,
        );
    }

    public function match(string $route): Route
    {
        return $this->routing[$route];
    }
}
